plugins/inventory/foreman.py

```diff
@@ -235,9 +235,15 @@ def _populate(self):
                 # create directly mapped groups
                 group_name = host.get('hostgroup_title', host.get('hostgroup_name'))
                 if group_name:
-                    group_name = to_safe_group_name('%s%s' % (self.get_option('group_prefix'), group_name.lower().replace(" ", "")))
-                    group_name = self.inventory.add_group(group_name)
-                    self.inventory.add_child(group_name, host_name)
+                    parent_name = None
+                    path = group_name.split('/')
+                    for i in range(1, len(path) + 1):
+                        gname = to_safe_group_name('%s%s' % (self.get_option('group_prefix'), '/'.join(path[0:i]).lower().replace(" ", "")))
```

Comment:
I had a bit of a hard time reading this loop and would propose an alternative writing:

```suggestion
                    group_label_parts = []
                    for part in group_name.split('/'):
                        group_label_parts.append(part.lower().replace(" ", ""))
                        gname = to_safe_group_name('%s%s' % (self.get_option('group_prefix'), '/'.join(group_label_parts)))
```